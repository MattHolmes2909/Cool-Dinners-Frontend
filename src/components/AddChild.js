import React, { useState } from "react";
import PostChild from "../requests/PostChild";
import Alert from "./Alert";
import "../styles/AddChild.css"

const AddChild = () => {
    const initialState = {
        fields: {
            childName: "",
            schoolClass: "1DS",
            foodOption: "none",
            allergies: "none"
        },
        alert: {
            message: "",
            isSuccess: false,
        },
    };

    const [fields, setFields] = useState(initialState.fields);
    const[alert, setAlert]= useState(initialState.alert);

    const handleAddChild = (event) => {
        event.preventDefault();
        setAlert({ message: "", isSuccess: false });
        PostChild(fields, setAlert);
    };
    
    const handleFieldChange = (event) => {
        setFields({ ...fields, [event.target.name]: event.target.value });
    };

    return (
        <div className="AddChild">
            <form onSubmit={handleAddChild} className="form">
                <p>Add Child</p>
                <div className="form-field">
                    <label htmlFor="title">
                        Child Name:
                        <input
                        name="childName"
                        value={fields.childName}
                        onChange={handleFieldChange}
                        />
                    </label>
                </div>
                <div className="form-child-class">
                    Class
                    <label htmlFor="child-class">
                        <select
                        id="schoolClass"
                        name="schoolClass"
                        value={fields.schoolClass}
                        onChange={handleFieldChange}
                        >
                            <option className="1DS" value="1DS">1DS</option>
                            <option className="1MH" value="1MH">1MH</option>
                            <option className="2AW" value="2AW">2AW</option>
                            <option className="2NM" value="2NM">2NM</option>
                        </select>
                    </label>
                </div>
                <div className="form-child-order">
                    Order
                    <label htmlFor="child-order">
                        <select
                        id="foodOption"
                        name="foodOption"
                        value={fields.foodOption}
                        onChange={handleFieldChange}
                        >
                            <option className="none" value="none">none</option>
                            <option className="pizza" value="pizza">pizza</option>
                            <option className="pasta" value="pasta">pasta</option>
                            <option className="fish" value="fish">fish</option>
                            <option className="quorn" value="quorn">quorn</option>
                        </select>
                    </label>
                </div>
                <div className="form-allergens">
                    Allergies
                    <label htmlFor="allergies">
                        <select
                        id="allergies"
                        name="allergies"
                        value={fields.allergies}
                        onChange={handleFieldChange}
                        >
                            <option className="none" value="none">none</option>
                            <option className="dairy" value="dairy">dairy</option>
                            <option className="wheat" value="wheat">wheat</option>
                            <option className="nuts" value="nuts">nuts</option>
                            <option className="fish" value="fish">fish</option>
                        </select>
                    </label>
                </div>
                <button type="submit" className="form-button">
                    Add
                </button>
                <Alert message={alert.message} success ={alert.isSuccess} />
            </form>
        </div>
    );
};

export default AddChild;
