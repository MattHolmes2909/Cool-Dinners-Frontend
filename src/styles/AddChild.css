.AddChild {
  animation: foodMove 100s linear infinite;
  background-image: url("../images/backgroundfood.png");
  background-position: bottom;
  background-repeat: repeat-x;
  background-size: 10%;
  height: 85vh;
}

.AddChildAdmin {
  animation: foodMove 100s linear infinite;
  background-image: url("../images/backgroundfood.png");
  background-position: bottom;
  background-repeat: repeat-x;
  background-size: 10%;
  height: 78vh;
}

.p-tag {
  font-size: 2em;
  margin: 0;
  padding: 0;
  text-align: center;
}

.childform {
  display: block;
  background: conic-gradient(
    #dc3e43 20%,
    #e9740e 20%,
    #e9740e 40%,
    #d2e367 40%,
    #d2e367 60%,
    #4cbf5d 60%,
    #4cbf5d 80%,
    #5383e9 80%,
    #5383e9 100%
  );
  background-size: 300%;
  background-position: -100px;
  border-radius: 25px;
  line-height: 3rem;
  margin: auto;
  padding: 1rem;
  text-align: center;
  width: 351px;
}

.childform-label {
  display: flex;
  flex-direction: row;
  justify-self: flex-end;
  text-align: center;
  line-height: 26px;
  margin-bottom: 10px;
  padding-left: 25px;
}

.childNameSelect {
  width: 40%;
  display: inline-block;
  border: 1px solid #ccc;
  border-radius: 4px;
}

.childNameSelect,
.schoolClassSelect,
.foodSelect,
.allergiesSelect,
.childform-button {
  text-align: center;
  text-align-last: center;
  -moz-text-align-last: center;
  font-size: 1em;
  height: 2em;
  flex: 0 0 250px;
  margin-left: 2em;
  font-family: "Otomanopee One", sans-serif;
}

.foodSelect,
.schoolClassSelect,
.allergiesSelect {
  cursor: pointer;
  border-radius: 12px;
}

.child-class-select {
  position: relative;
  right: 2.2rem;
}

.teacher-class {
  font-size: 1.2rem;
  margin-left: 1.9rem;
}

.childform-button {
  background: rgb(96, 52, 153);
  border: 0;
  border-radius: 12px;
  color: white;
  padding: 10px 30px 35px;
  margin-bottom: 1em;
  margin-top: 1em;
  margin-right: 40px;
  font-family: "Bebas Neue", sans-serif;
  font-size: 1.4em;
}

.childform-button:hover {
  animation: add-hover 1s forwards;
  cursor: pointer;
}

@keyframes add-hover {
  from {
    background: rgb(103, 47, 176);
  }
  to {
    background: rgb(134, 97, 181);
  }
}
