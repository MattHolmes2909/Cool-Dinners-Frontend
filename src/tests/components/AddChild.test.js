import React from "react";
import { render } from "@testing-library/react";
import AddChild from "../../components/AddChild";

describe("AddChild", () => {
    it("renders correctly", () => {
        const { asFragment } = render (<AddChild />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders input correctly", () => {
        const { getByTestId } = render(<AddChild />);
        const input = getByTestId("form-title-input-child");
        expect(input).toBeTruthy();
    });

    it("Renders button correctly", () => {
        const { getByTestId } = render(<AddChild />);
        const button = getByTestId("form-button-child");
        expect(button).toBeTruthy();
    });
});