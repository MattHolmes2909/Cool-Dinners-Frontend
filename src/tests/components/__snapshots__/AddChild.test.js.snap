// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddChild renders correctly 1`] = `
<DocumentFragment>
  <div
    class="AddChild"
  >
    <form
      class="form-AddChild"
    >
      <p
        class="addChild"
      >
        Add Child
      </p>
      <div
        class="form-field-AddChild"
      >
        <label
          for="title-AddChild"
        >
          Child Name:
          <input
            data-testid="form-title-input-child"
            name="childName"
            placeholder="Your Name Here..."
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        class="form-child-class"
      >
        Class
        <label
          for="child-class"
        >
          <select
            id="schoolClass"
            name="schoolClass"
          >
            <option
              class="1DS"
              value="1DS"
            >
              1DS
            </option>
            <option
              class="1MH"
              value="1MH"
            >
              1MH
            </option>
            <option
              class="2AW"
              value="2AW"
            >
              2AW
            </option>
            <option
              class="2NM"
              value="2NM"
            >
              2NM
            </option>
          </select>
        </label>
      </div>
      <div
        class="form-child-order"
      >
        Order
        <label
          for="child-order"
        >
          <select
            id="foodOption"
            name="foodOption"
          >
            <option
              class="none"
              value="none"
            >
              none
            </option>
            <option
              class="pizza"
              value="pizza"
            >
              pizza
            </option>
            <option
              class="pasta"
              value="pasta"
            >
              pasta
            </option>
            <option
              class="fish"
              value="fish"
            >
              fish
            </option>
            <option
              class="quorn"
              value="quorn"
            >
              quorn
            </option>
          </select>
        </label>
      </div>
      <div
        class="form-allergens"
      >
        Allergies
        <label
          for="allergies"
        >
          <select
            id="allergies"
            name="allergies"
          >
            <option
              class="none"
              value="none"
            >
              none
            </option>
            <option
              class="dairy"
              value="dairy"
            >
              dairy
            </option>
            <option
              class="wheat"
              value="wheat"
            >
              wheat
            </option>
            <option
              class="nuts"
              value="nuts"
            >
              nuts
            </option>
            <option
              class="fish"
              value="fish"
            >
              fish
            </option>
          </select>
        </label>
      </div>
      <button
        class="form-button-child"
        data-testid="form-button-child"
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
</DocumentFragment>
`;
